seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/hdfs/HdfsParquetTransactionStateFileWriter.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.hadoop.ParquetFileWriter;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.hadoop.util.HadoopOutputFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HdfsParquetTransactionStateFileWriter extends AbstractTransactionStateFileWriter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HdfsParquetTransactionStateFileWriter.class);
+    private Map<String, ParquetWriter<GenericRecord>> beingWrittenWriter;
+
+    public HdfsParquetTransactionStateFileWriter(@NonNull SeaTunnelRowType seaTunnelRowTypeInfo,
+                                                 @NonNull TransactionFileNameGenerator transactionFileNameGenerator,
+                                                 @NonNull PartitionDirNameGenerator partitionDirNameGenerator,
+                                                 @NonNull List<Integer> sinkColumnsIndexInRow, @NonNull String tmpPath,
+                                                 @NonNull String targetPath,
+                                                 @NonNull String jobId,
+                                                 int subTaskIndex,
+                                                 @NonNull FileSystem fileSystem) {
+        super(seaTunnelRowTypeInfo, transactionFileNameGenerator, partitionDirNameGenerator, sinkColumnsIndexInRow, tmpPath, targetPath, jobId, subTaskIndex, fileSystem);
+        beingWrittenWriter = new HashMap<>();
+    }
+
+    @Override
+    public void write(@NonNull SeaTunnelRow seaTunnelRow) {
+        String filePath = getOrCreateFilePathBeingWritten(seaTunnelRow);
+        ParquetWriter<GenericRecord> writer = getOrCreateWriter(filePath);
+        Schema schema = buildSchemaWithRowType();
+        GenericRecordBuilder recordBuilder = new GenericRecordBuilder(schema);
+        sinkColumnsIndexInRow.forEach(index -> {
+            if (seaTunnelRowTypeInfo.getFieldType(index).equals(BasicType.STRING_TYPE)) {
+                recordBuilder.set(seaTunnelRowTypeInfo.getFieldName(index), seaTunnelRow.getField(index).toString());
+            } else {
+                recordBuilder.set(seaTunnelRowTypeInfo.getFieldName(index), seaTunnelRow.getField(index));
+            }
+        });
+        GenericData.Record record = recordBuilder.build();
+        try {
+            writer.write(record);
+        } catch (IOException e) {
+            LOGGER.error("Write data to file [{}] error", filePath);
+            throw new RuntimeException(e);
```

Comment:
If an exception needs to be thrown, then there is no need to use Log.error

Reply:
done